### PR TITLE
Ensure the map does not freeze when no structures are returned

### LINKED ISSUE #1100: 400 Error freezes map view when no structures are returned on Thailand Production
The Thailand production server fails to load the map to view if no structures are returned from the data warehouse. This 400 error is marked as a bad request, but it's just the lack of results from Superset. The cited jurisdiction does not have any structures in the Reveal transactional database. jurisdiction_id = 398f1463-0be3-4e6b-9c9e-31ee755e1546

Symptoms:
Spinning map in the web UI:
image

When I dig into the network, I see a 400 error on slice 226: https://superset.reveal-th.smartregister.org/superset/slice_json/226?form_data={%22adhoc_filters%22:[{%22clause%22:%22WHERE%22,%22expressionType%22:%22SIMPLE%22,%22comparator%22:%22398f1463-0be3-4e6b-9c9e-31ee755e1546%22,%22operator%22:%22==%22,%22subject%22:%22jurisdiction_id%22}],%22row_limit%22:15000}

Rowcount returned = 0, so it's not a bad request, it's just that no results were returned
image

I copied the query into Superset SQL builder and there are no results returned:
image
```
SELECT id,
jurisdiction_id
FROM structures
WHERE jurisdiction_id = '398f1463-0be3-4e6b-9c9e-31ee755e1546'
LIMIT 15000;
```
No structures are returned from the Reveal database:
https://reveal-th.smartregister.org/opensrp/rest/location/findByProperties?is_jurisdiction=false&return_geometry=false&properties_filter=parentId:398f1463-0be3-4e6b-9c9e-31ee755e1546

Expected Behavior:
1. the map view renders regardless of the 400 error, it just doesn't show any structures
2. It would be valuable if our handling of the Superset 0 results didn't return a 400 error, just 0 results.

### PR DESCRIPTION
**Pull request by kelvin-muchiri**
Closes #1100 

The old Gisida used previously had the map fail to display incase no structures were returned. A test to ensure this does not happen in GisidaLite is added

### CODE REVIEW

File: src/containers/pages/FocusInvestigation/map/active/tests/indexCases.test.tsx [hunk 1]
```diff
@@ -153,4 +156,52 @@ describe('containers/pages/FocusInvestigation/activeMap', () => {
...
+    const componentProps: any = wrapper.find('SingleActiveFIMap').props();
+
+    expect(componentProps.plan.id).toEqual('dbd9851f-2548-5aaa-8267-010897f98f45');
+    expect(componentProps.jurisdiction.jurisdiction_id).toEqual('3951');
+    expect(componentProps.pointFeatureCollection.features.length).toEqual(95);
+    expect(componentProps.polygonFeatureCollection.features.length).toEqual(7);
+    expect(componentProps.currentPointIndexCases.features.length).toEqual(1);
+    expect(componentProps.currentPolyIndexCases.features.length).toEqual(0);
+    expect(componentProps.historicalPointIndexCases.features.length).toEqual(0);
+    expect(componentProps.historicalPolyIndexCases.features.length).toEqual(2);
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/map/active/tests/indexCases.test.tsx:201]: Lets add a test where all these are empty i.e. everything apart from the jurisdiction boundary is absent

File: src/containers/pages/FocusInvestigation/map/active/tests/supersetFixtures/jurisdictionsNoGeojson.json [hunk 2]
```diff
@@ -0,0 +1,328 @@
...
+      {
+        "jurisdiction_id": "3951"
+      },
+      {
+        "jurisdiction_id": "3951"
+      },
+      {
+        "jurisdiction_id": "3951"
+      },
+      {
```
Comment (moshthepitt) [inline src/containers/pages/FocusInvestigation/map/active/tests/supersetFixtures/jurisdictionsNoGeojson.json:64]: Why are these repeated this way?

Comment (kelvin-muchiri) [reply src/containers/pages/FocusInvestigation/map/active/tests/supersetFixtures/jurisdictionsNoGeojson.json]: Its also repeated in the other fixture, but I've refactored to only pass the array

**Comment by p-netm** [comment]
@kelvin-muchiri 
please look into this red text:
``` Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default ```

File: src/containers/pages/FocusInvestigation/map/active/tests/indexCases.test.tsx [hunk 3]
```diff
@@ -142,7 +145,145 @@ describe('containers/pages/FocusInvestigation/activeMap', () => {
...
+    const wrapper = mount(
+      <Provider store={store}>
+        <Router history={history}>
+          <ConnectedMapSingleFI {...props} />
+        </Router>
+      </Provider>
+    );
+
+    await new Promise(resolve => setImmediate(resolve));
+    wrapper.update();
```
Comment (p-netm) [inline src/containers/pages/FocusInvestigation/map/active/tests/indexCases.test.tsx:190]: lets also please wrap these in async `act`

**Comment by kelvin-muchiri** [comment]
> @kelvin-muchiri
> please look into this red text:
> `Warning: React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default`

Where is this?

**Comment by p-netm** [comment]
> > please look into this red text:
> 
> @p-netm Where is this?

once you run the test

**Comment by kelvin-muchiri** [comment]
> once you run the test

I am not getting the error, circleci has also returned successfully

**Comment by p-netm** [comment]
> > once you run the test
> 
> @p-netm I am not getting the error, circleci has also returned successfully

its not an error that results in failed tests. it howver shows there is something out of the norm happening inside the tests.

**Review by p-netm — APPROVED** [review]

**Review by moshthepitt — APPROVED** [review]
